app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CompaniesController < ApplicationController
+
+	before_action :authenticate_user!
+
+	def new
+
+	end
+
+	def create
```

Comment:
This method is doing lots of thing that I don't understand. It simply doesn't follow SRP. Refactor it.